### DISCUSSION TITLE
Would you add more documents for callbacks?

### DISCUSSION THREAD

**Discussion by Streamlet** [microsoft/msquic#4698] [General]
The documents for quic functions is very detail, but there is few for callbacks.
Would you add more documents for each parameters of each kind of callbacks?

**Comment by Streamlet** [comment]
e.g.,

# Listener callbacks

## QUIC_LISTENER_EVENT_NEW_CONNECTION

### Listener
...
### Context
...
### Event
QUIC_LISTENER_EVENT structure
#### Type
always be QUIC_LISTENER_EVENT_NEW_CONNECTION
#### NEW_CONNECTION
For QUIC_LISTENER_EVENT_NEW_CONNECTION, members in the union except NEW_CONNECTION are invalid.
* Info: ...
  * QuicVersion: ...
  * LocalAddress: ...
  * RemoteAddress: ...
  * CryptoBufferLength: ...
  * ClientAlpnListLength: ...
  * ServerNameLength: ...
  * NegotiatedAlpnLength: ...
  * CryptoBufferLength: ...
  * CryptoBuffer: ...
  * ClientAlpnListLength: ...
  * ClientAlpnList: ...
  * NegotiatedAlpnLength: ...
  * NegotiatedAlpn: ...
  * ServerNameLength: ...
  * ServerName: ...
* Connection: ...

### Return
Application should always return QUIC_STATUS_SUCCESS.

### Remark
...
(Write some important things for internal implement, to help applications to use the API correctly)
(How to co-work with other APIs or callbacks? Resources need to be cleaned later? ...)

## QUIC_LISTENER_EVENT_STOP_COMPLETE
...

# Connection callbacks
...

# Stream callbacks
...

**Reply by nibanks** [reply]
Created https://github.com/microsoft/msquic/issues/4700 to track this officially.
